docs/messaging.rst

```diff
@@ -859,7 +859,9 @@ multiple cases:
 
 The client sends a shutdown request to the kernel, and once it receives the
 reply message (which is otherwise empty), it can assume that the kernel has
-completed shutdown safely.
+completed shutdown safely.  The request can be sent on either the `control` or
+`shell` channels.  The reply is made both to the requesting channel and to
+the `iopub` channel.
```

Comment:
It is what the IPython kernel does, and I think it was added during some of the multi-frontend stuff with the QtConsole (shutdown logic is a tangled mess over there). Sending the reply instead of a notification or status message was definitely a mistake.

I'll look at the QtConsole code and see if I can figure out what it's used for. If it behaves okay with kernels that don't implement this undocumented behavior, I would prefer updating the IPython kernel to match the spec rather than the spec to match this icky behavior from the kernel.